samples/grids/grid/finjs/src/index.tsx

```diff
@@ -0,0 +1,400 @@
+const changeNegative = (rowData: any): boolean => rowData['changeP'] < 0 && rowData['changeP'] > -1;
+const changePositive = (rowData: any): boolean => rowData['changeP'] > 0 && rowData['changeP'] < 1;
+const strongPositive = (rowData: any): boolean => rowData['changeP'] >= 1;
+const strongNegative = (rowData: any): boolean => rowData['changeP'] <= -1;
+const trends = {
+  changeNeg: changeNegative,
+  changePos: changePositive,
+  negative: negative,
+  positive: positive,
+  strongNegative: strongNegative,
+  strongPositive: strongPositive
+};
+
+const trendsChange = {
+  changeNeg2: changeNegative,
+  changePos2: changePositive,
+  strongNegative2: strongNegative,
+  strongPositive2: strongPositive
+};
+export default function Sample() {
+  const gridRef = useRef<IgrGrid>(null);
+
+function priceTemplate(ctx: {dataContext: IgrCellTemplateContext}) {
+  const cell = ctx.dataContext.cell;
+  const rowData = gridRef.current.getRowData(cell.id.rowID);
+  const icon = trends.positive(rowData) ? "trending_up" : "trending_down";
+  const value = cell.value.toFixed(4);
+  return <>
+  <div className="finjs-icons">
+  <span>${value}</span>
+  <IgrIcon name={icon} collection="material"></IgrIcon>
+  </div>
+  </>
+}
+
+function chartBtnTemplate(ctx: {dataContext: IgrCellTemplateContext}) {
+  const cell = ctx.dataContext.cell;
+  const rowData = gridRef.current.getRowData(cell.id.rowID);
+  return<> <span onClick={e => openDialogForRow(e, rowData)}><IgrIcon name="insert_chart" collection="material" variant="contained" size="small">
+  </IgrIcon></span>
+  </>
+}
+
+const iconForUpdate = useRef<IgrIcon>(null);
+const iconForStop = useRef<IgrIcon>(null);
+const iconForChart = useRef<IgrIcon>(null);
+
+const startButton = useRef<IgrButton>(null);
+const stopButton = useRef<IgrButton>(null);
+const chartButton = useRef<IgrButton>(null);
+
+const chartRef = useRef<IgrCategoryChart>(null);
+const dialogRef = useRef<IgrDialog>(null);
+const toastRef = useRef<IgrToast>(null);
+
+useEffect(() => {
+  if (iconForUpdate?.current) {
+    iconForUpdate.current.registerIconFromText(
+      "update",
+      updateIcon,
+      "material"
+    );
+  }
+  if (iconForStop?.current) {
+    iconForStop.current.registerIconFromText(
+      "stop",
+      stopIcon,
+      "material"
+    );
+  }
+  if (iconForChart?.current) {
+    iconForChart.current.registerIconFromText(
+      "insert_chart",
+      chartIcon,
+      "material"
+    );
+  }
+}, []);
+
+
+const groupingEnabled = true;
+const toolbarEnabled = true;
+const [recordsCount, setRecordsCount] = useState(1000)
+const [frequency, setFrequency] = useState(500);
+let _timer:any;
+
+function startUpdate() {
+  _timer = setInterval(() => {
+      gridRef.current.data = FinancialData.updateAllPrices(data);
+  }, frequency);
+  startButton.current.disabled = true;
+  stopButton.current.disabled = false;
+  chartButton.current.disabled = true;
+}
+
+function stopUpdate() {
+  clearInterval(_timer);
```

Comment:
`_times` becomes undefined when Live data is enabled and you change the `recordsCount` or `frequency`.
I think the `_timer` should be saved in a state as well.